### PR TITLE
git-clean: add page

### PR DESCRIPTION
**Pull request by eliotsykes**

### CODE REVIEW

**Comment by waldyrious** [comment]
@agnivade maybe let's merge and add the options in a separate PR afterwards?

**Comment by eliotsykes** [comment]
@waldyrious @agnivade Please do

**Comment by waldyrious** [comment]
@eliotsykes we'd have preferred if you had mentioned that you preferred not doing that. Please let us know next time, ok?

**Comment by eliotsykes** [comment]
Apologies - no harm intended, it slid off my radar. Many thanks for all your work on tldr and happy holidays!

**Comment by waldyrious** [comment]
Thanks for clarifying @eliotsykes. We've had other PRs where people stopped responding out of what seemed to be annoyance at our admittedly exhaustive review process. I could have misinterpreted that, as I clearly did here. Apologies for that 😳

**Comment by eliotsykes** [comment]
No apology necessary - I really appreciate the project, and understand how easy it is for comments to come across in an unintended way. I read back my original response and its way too short and wish I'd worded it better - I'm sorry for the miscommunication. This is a fantastic project and thank you again 😃

**Comment by waldyrious** [comment]
Glad it's all cleared up now for all sides. Let's wait for @agnivade's reaction :)

**Comment by agnivade** [comment]
Since @eliotsykes is back again, I would prefer that we resolve the comments and then merge.

**Comment by eliotsykes** [comment]
@agnivade If you want to provide the exact text to add, please do and I'll add a commit with it in, based on your previous comments:

> I would also add -f and -x.

> -f because usually requireForce is set to true. So git clean will refuse to delete anything unless you pass -f

> -x because it ignores the .gitignore rules. Very helpful in deleting build directories in cases when you want to start with a clean build.

**Comment by agnivade** [comment]
You might need to simplify the language a bit, but this is what I think should be the basic message -
-f - Usually the git variable `clean.requireForce` is `true`. So pass this option to actually delete.
-x - This ignores files skipped by .gitignore. Helpful in removing build directories

File: pages/common/git-clean.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+- Interactively delete files that are not tracked by git:
+
+`git clean -i`
+
+- Shows what files would be deleted without actually deleting them:
+
+`git clean --dry-run`
+
+- Force delete files that are not tracked by git:
```
Comment (waldyrious) [inline pages/common/git-clean.md]: Perhaps "Force deletion" would work better.

File: pages/common/git-clean.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+`git clean -f`
+
+- Delete untracked files, including ignored files in `.gitignore` and `$GIT_DIR/info/exclude`:
```
Comment (waldyrious) [inline pages/common/git-clean.md]: I'd suggest using `.git/info.exclude` -- let's use the default and avoid introducing unnecessary complexity :)

File: pages/common/git-clean.md (hunk 1 above)
Comment (agnivade) [inline pages/common/git-clean.md]: Don't want to raise grammar issues here, but I kinda like 'Force delete'. :)

Comment (waldyrious) [inline pages/common/git-clean.md]: To use "delete" as the verb, it should be "force-delete" or "forcefully delete".

**Comment by agnivade** [comment]
@eliotsykes - any updates on this ?

**Comment by agnivade** [comment]
Please fix the travis errors. We should be good after that.

**Comment by eliotsykes** [comment]
I appreciate the work that everyone does with TLDR and I've enjoyed making contributions up until this one. 

In the interests of hoping this helps the project to grow strength from strength, I'd like to say the experience of contibutions has become too onerous for me. 

For a documentation project, the build automation feels too much and may be getting in the way of good work that could be helping more people and being more welcoming to contributors. Exhibit A:

> TLDR104 Example descriptions should prefer infinitive tense (e.g. write) over present (e.g. writes) or gerund (e.g. writing)

I hope TLDR will consider returning to a more lightweight process and PR review in the future.

**Comment by agnivade** [comment]
Hey @eliotsykes, sorry that you feel that way. The build automation is mainly to maintain an uniform structure for the pages. I am seeing grammar related errors like this for the first time. Probably @rubenvereecken can comment further on the rule.

The maintainers work hard to make the project as accessible as possible. Certain automations make our lives easy. I have been chasing after https://github.com/tldr-pages/tldr/issues/1066 for a long time. And been unsuccessful.

Anyways, thanks a lot for this PR. Don't worry about it. When I find some time, I will fix the grammar error and include it in the repo so that your commit history is maintained.

**Comment by waldyrious** [comment]
@eliotsykes thank you for the candid feedback. We care deeply about making tldr-pages a welcoming project for all contributors, and we have made some effort in that goal, with more to come -- you can track the progress of these efforts here: https://github.com/tldr-pages/tldr/projects/3.

We do realize we're quite a long way from the ideal contribution experience, and that we often make contributors jump over too many hoops for the sake of maintaining a consistent content within the repo and complying with the guidelines we have (collectively) defined. For that we're sorry, and totally understand the frustration that results from it.

That said, we hope it's clear that we're concerned about this issue, and actively working (within our limited availability) to improve on the current state of things. We'd be thrilled to count on your help to achieve that, even if just pointing out places where our efforts could be made more visible so people are aware of them and can help pushing those goals forward -- but just providing honest feedback and constructive criticism is already helpful, so thank you for that :) we hope to be able to gain your trust in our ability to welcome your contributions soon.

**Comment by eliotsykes** [comment]
Hi @waldyrious - thank you for taking the time to write. I continue to benefit from using this project for which I'm also grateful, and its good to hear the contribution experience is being improved. If I find an opportunity to contribute in the future I'll try again and try to provide constructive feedback.

**Comment by waldyrious** [comment]
Thanks for understanding :) hopefully next time things will run smoother.

**Comment by waldyrious** [comment]
A Travis failure for such a minor issue (verb tense) is definitely an unacceptably aggressive feedback for contributors, especially if we ask them to fix it themselves. It makes sense when the change could be subject to editorial discretion (wording, etc.), but uncontroversial minor fixes should be done by maintainers. I'll make a proposal (hopefully tonight) for a maintainership guide that takes this in consideration.

As for this PR, I'll reopen it and make the change myself, so it can be merged.

**Comment by waldyrious** [comment]
@eliotsykes I don't seem to be able to edit this file -- do you see a checkbox saying "allow edits by maintainers"? If so, please check it so we can take care of the PR.

**Comment by eliotsykes** [comment]
@waldyrious Yep the checkbox was on the bottom of the right-hand column on GitHub, below the participants section. Its checked now 👍

**Comment by waldyrious** [comment]
@agnivade please take a look :) I think the requests in your review should be left to a separate PR (unless you'd like to take a stab at them). Maybe copy them to a new issue, so that they remain tracked?

**Comment by agnivade** [comment]
If you are talking about [this](https://github.com/tldr-pages/tldr/pull/1157#pullrequestreview-9247769), this is already taken care of no ?

**Comment by waldyrious** [comment]
Oops! My bad, I misread the diff. Sounds like this is ready to go, then :D

**Comment by agnivade** [comment]
Ah my bad, I should have approved too.

**Pull request merged**